hub/core/storage/tests/test_storage_provider.py

```diff
@@ -126,12 +127,13 @@ def test_cache(storage):
 
 @parametrize_all_storages
 def test_pickling(storage):
-    with CliRunner().isolated_filesystem():
-        FILE_1 = f"{KEY}_1"
-        storage[FILE_1] = b"hello world"
-        assert storage[FILE_1] == b"hello world"
-        pickle_file = open("storage_pickle", "wb")
-        pickle.dump(storage, pickle_file)
-        pickle_file = open("storage_pickle", "rb")
-        unpickled_storage = pickle.load(pickle_file)
-        assert unpickled_storage[FILE_1] == b"hello world"
+    if isinstance(storage, MemoryProvider):
+        # skip pickling test for memory provider as the actual data isn't pickled for it
+        return
```

Comment:
nooooo this is what the fixtures are for >:-( 

can you please instead base this branch onto `update/test-fixtures` (im about to merge it right now), i added a new fixture `@enabled_persistent_dataset_generators` that doesn't contain memory datasets.

when you base off that branch, inside `hub.tests.dataset_fixtures` you will see `@enabled_persistent_dataset_generators`, this is for *dataset generators*. 

this won't work for this case, because you are using the `storage` fixture, so you should create a new parametrization for this.

you may do this by going into `hub.tests.storage_fixtures` and copying `enabled_storages` (defined in there), and create a new one right below it called `enabled_persistent_storages`, excluding the memory string entry.

i would do this myself, but it would be nice for everyone to get acquainted with creating new parametrizes/fixtures so i'm not the only one doing it.